src/plugins/controls/public/control_group/external_api/control_group_renderer.tsx

```diff
@@ -6,136 +6,194 @@
+          (await getCreationOptions?.(
+            getDefaultControlGroupRuntimeState(),
+            controlGroupStateBuilder
+          )) ?? {};
+        updateInput({
+          ...initialState,
+          editorConfig,
+        });
+        const state = {
+          ...omit(initialState, ['initialChildControlState', 'ignoreParentSettings']),
+          editorConfig,
+          controlStyle: initialState?.labelPosition,
+          panelsJSON: JSON.stringify(initialState?.initialChildControlState ?? {}),
+          ignoreParentSettingsJSON: JSON.stringify(initialState?.ignoreParentSettings ?? {}),
         };
-        const { initialInput, settings, fieldFilterPredicate } =
-          (await getCreationOptions?.(getDefaultControlGroupInput(), controlGroupInputBuilder)) ??
-          {};
-        const newControlGroup = (await factory?.create(
-          {
-            id,
-            ...getDefaultControlGroupInput(),
-            ...initialInput,
-          },
-          undefined,
-          {
-            ...settings,
-            lastSavedInput: {
-              ...getDefaultControlGroupPersistableInput(),
-              ...pick(initialInput, persistableControlGroupInputKeys),
-            },
-          },
-          fieldFilterPredicate
-        )) as ControlGroupContainer;
 
-        if (canceled) {
-          newControlGroup.destroy();
-          controlGroup?.destroy();
-          return;
+        if (!cancelled) {
+          setSerializedState(state as ControlGroupSerializedState);
         }
-
-        if (controlGroupDomRef.current) {
-          newControlGroup.render(controlGroupDomRef.current);
-        }
-        setControlGroup(newControlGroup);
-        destroyControlGroup = () => newControlGroup.destroy();
       })();
       return () => {
-        canceled = true;
-        destroyControlGroup?.();
+        cancelled = true;
       };
       // exhaustive deps disabled because we want the control group to be created only on first render.
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
-    useEffect(() => {
-      if (!controlGroup) return;
-      if (
-        (timeRange && !isEqual(controlGroup.getInput().timeRange, timeRange)) ||
-        !compareFilters(controlGroup.getInput().filters ?? [], filters ?? []) ||
-        !isEqual(controlGroup.getInput().query, query)
-      ) {
-        controlGroup.updateInput({
-          timeRange,
-          query,
-          filters,
-        });
-      }
-    }, [query, filters, controlGroup, timeRange]);
-
-    return <div ref={controlGroupDomRef} />;
+    return !serializedState ? null : (
+      <ReactEmbeddableRenderer<
+        ControlGroupSerializedState,
+        ControlGroupRuntimeState,
+        ControlGroupApi
+      >
+        key={regenerateId} // this key forces a re-mount when `updateInput` is called
+        maybeId={id}
+        type={CONTROL_GROUP_TYPE}
+        getParentApi={() => ({
+          reload$,
+          dataLoading$,
```

Comment:
this needs to be called `dataLoading` because that is how its keyed in the interface packages/presentation/presentation_publishing/interfaces/publishes_data_loading.ts.

We need to spend some time cleaning up these keys to ensure consistent naming convensions.

Reply:
Ohh man yeah, we need consistency so bad here 😟 Thanks for catching that - fixed in b8f6f4409243acc210108eb5583cdb611c08f160